firmware/src/network_model.c

```diff
@@ -377,59 +329,50 @@ int NetworkModel_SetZeroconfMode(const RDMHeader *header,
 
 int NetworkModel_GetCurrentAddress(const RDMHeader *header,
                                    const uint8_t *param_data) {
-  unsigned int index = 0;
-  int r = CheckInterfaceRequest(header, param_data, &index);
-  if (r >= 0) {
-    return r;
+  const int index = LookupIndex(ExtractUInt32(param_data));
+  if (index < 0) {
+    return RDMResponder_BuildNack(header, NR_DATA_OUT_OF_RANGE);
   }
 
-  ReturnUnlessUnicast(header);
   InterfaceState *interface = &g_network_model.interfaces[index];
 
-  unsigned int offset = sizeof(RDMHeader);
-  memcpy(g_rdm_buffer + offset, param_data, INTERFACE_ID_SIZE);
-  offset += INTERFACE_ID_SIZE;
-  g_rdm_buffer[offset++] = UInt32Byte0(interface->current_ip);
-  g_rdm_buffer[offset++] = UInt32Byte1(interface->current_ip);
-  g_rdm_buffer[offset++] = UInt32Byte2(interface->current_ip);
-  g_rdm_buffer[offset++] = UInt32Byte3(interface->current_ip);
-  g_rdm_buffer[offset++] = interface->current_netmask;
+  uint8_t *ptr = g_rdm_buffer + sizeof(RDMHeader);
+  memcpy(ptr, param_data, INTERFACE_ID_SIZE);
```

Comment:
We reuse the interface id from the request so that we don't have to keep another variable around.  This reduces the amount of code on line 332